include/swift/Runtime/CrashInfo.h

```diff
@@ -18,22 +18,52 @@
 #ifndef SWIFT_CRASHINFO_H
 #define SWIFT_CRASHINFO_H
 
-#include "SwiftStdint.h"
+#include <inttypes.h>
 
 #ifdef __cplusplus
 namespace swift {
+namespace runtime {
+namespace backtrace {
 extern "C" {
 #endif
 
 struct CrashInfo {
-  __swift_uint64_t crashing_thread;
-  __swift_uint64_t signal;
-  __swift_uint64_t fault_address;
-  __swift_uint64_t mctx;
+  uint64_t crashing_thread;
+  uint64_t signal;
+  uint64_t fault_address;
+
+#ifdef __APPLE__
+  uint64_t mctx;
+#elif defined(__linux__)
+  uint64_t thread_list;
+#endif
+};
+
+#ifdef __linux__
+
+struct memserver_req {
+  uint64_t addr;
+  uint64_t len;
 };
 
+struct memserver_resp {
+  uint64_t addr;
+  int64_t  len;
+  /* Then len bytes of data */
+};
+
+struct thread {
```

Comment:
This would benefit from a doc comment that clarifies what `next` and `uctx` are for. I'd assume `tid` refers to "thread id", but a doc comment would confirm that too.

Reply:
As is this, I think (#66334).